### PR TITLE
restrict sharepoint test backups to test folders

### PR DESCRIPTION
**Pull request by ryanfkeepers**
To avoid over-production of data, and thus longer
CI times, any sharepoint integration test that can scope
library usage down to only the "test" folders should do so.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :robot: Supportability/Tests

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/connector/data_collections_test.go [hunk 1]
```diff
@@ -334,7 +335,7 @@ func (suite *ConnectorCreateSharePointCollectionIntegrationSuite) TestCreateShar
 	)
 
 	sel := selectors.NewSharePointBackup(siteIDs)
-	sel.Include(sel.LibraryFolders([]string{"foo"}, selectors.PrefixMatch()))
+	sel.Include(sel.LibraryFolders([]string{"zzazil"}, selectors.PrefixMatch()))
```
Comment (pandeyabs) [inline src/internal/connector/data_collections_test.go:338]: What does this magic keyword do?

File: src/pkg/selectors/testdata/sharepoint.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package testdata
+
+import "github.com/alcionai/corso/src/pkg/selectors"
+
+// BackupFolderScope is the standard folder scope that should be used
+// in integration backups with sharepoint.
+func BackupFolderScope(sel *selectors.SharePointBackup) []selectors.SharePointScope {
```
Comment (ashmrtn) [inline src/pkg/selectors/testdata/sharepoint.go:7]: nit: should prefix this as `SharePointBackupFolderScope` as other services may need the same sort of function in the future

File: src/internal/connector/data_collections_test.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/connector/data_collections_test.go:338]: That magic keyword reminds me that I need to come back and investigate a change in that location.  Usually.  Eventually.

**Comment by ryanfkeepers** [comment]
@pandeyabs @ashmrtn thanks for the comments.  This is ready for a final review.

**Review by pandeyabs — APPROVED** [review]

**Pull request merged**